tests/pytest_framework/src/syslog_ng_config/statement_group.py

```diff
@@ -22,13 +22,24 @@
 #############################################################################
 
 from src.common.random_id import get_unique_id
-
+import functools
+from src.common.operations import cast_to_list
 
 class StatementGroup(object):
-    def __init__(self, group_type):
-        self.__group_type = group_type
-        self.__group_id = "%s_%s" % (group_type, get_unique_id())
+    def __init__(self, statements):
+        self.__group_type = self.__calculate_group_type(cast_to_list(statements))
+        self.__group_id = "%s_%s" % (self.__group_type, get_unique_id())
         self.__statements = []
+        if statements:
+            self.update_group_with_statements(cast_to_list(statements))
```

Comment:
I would not put self.update_group_with_statements() in to constructor. From the caller side I would not expect this.
I would call it separately after the StatementGroup() is created. I understand it is one more step which we can avoid.

Reply:
Sorry I cannot understand this one. Why does it look unexpected from the caller site? What would be the definition/short explanation of `StatementGroup`?

This is how I think about it: `StatementGroup` is a container type that is intended to store statements. With that, it is very close in feeling to a regular list, just with a few extras. As I do not feel strange to initialize a regular list through constructor, it feels analogous here too.

But I know it is not how it is implemented, and this might be the source of disagreement. Now to mention it, I feel strange about the implementation.
StatementGroup should be a list(-ish) from ducktype point of view, with the add/remove (update_group_with_statement/remove_statement). But you cannot access the content as you would do with a list, because the content is exposed through a property.

What do you think about making the list analogue more explicit? I can reimplement it by inheriting from `list` directly. That would remove some code: the add/remove methods are not needed anymore. And also would have a lot of exciting benefit: for example you could write statement_group[0], or can use pop, insert, clear, copy, etc. Also for the iteration, it would look very neat to write `for statement in statement_group:` compared to `for statement in statement_group.statements`.

What do you think?

Reply:
I understand the benefits of inheritance from list object. 
But we should be able to update (for now we have this with: update_group_with_statements()) the statement_group list with more than one statements in one turn. 
Originally lists has append(), but it is not accepts more than one items. We would need such function (will be this understandable?):
```
def append_items(self, items):
   for item in items:
      self.append(item)
```

Reply:
If we will not lose functionality from StatementGroup() than +1 from my side

Reply:
Ok, I will do so. The builtin list contains an `extend` method that adds many elements based on an iterator that should be available for us. So we would not even need `append_items`.